chips/earlgrey/src/timer.rs

```diff
@@ -94,19 +94,14 @@ impl time::Time for RvTimer<'_> {
     type Ticks = Ticks64;
 
     fn now(&self) -> Ticks64 {
-        // RISC-V has a 64-bit counter but you can only read 32 bits
-        // at once, which creates a race condition if the lower register
-        // wraps between the reads. So the recommended approach is to read
-        // low, read high, read low, and if the second low is lower, re-read
-        // high. -pal 8/6/20
-        let first_low: u32 = self.registers.value_low.get();
-        let mut high: u32 = self.registers.value_high.get();
-        let second_low: u32 = self.registers.value_low.get();
-        if second_low < first_low {
-            // Wraparound
-            high = self.registers.value_high.get();
-        }
-        Ticks64::from(((high as u64) << 32) | second_low as u64)
+        let mtimer = MachineTimer::new(
+            &self.registers.compare_low,
+            &self.registers.compare_high,
+            &self.registers.value_low,
+            &self.registers.value_high,
+        );
```

Comment:
Would it make more sense to just call `MachineTimer::new()` once and save it as a field on `rv_timer`, which is then accessed from the other functions?

Reply:
Yes it would, but I couldn't figure out how to do that while keeping lifetimes happy.